int32 bd_sig
int32 ad_sig
int32 wpd_sig
int32 table_sig
